## **Los indicadores de capacidad estadística brindan información sobre varios aspectos de los sistemas estadísticos nacionales de los países en desarrollo, incluido un indicador general de capacidad estadística a nivel nacional.**

Link: https://databank.worldbank.org/source/statistical-capacity-indicators

                        Reporte de calidad de datos para la carpeta statistical_capacity_indicators

In [6]:
import pandas as pd
import numpy as np

In [4]:
data1 = pd.read_csv('1190a961-8921-49dc-ac16-6de9f797c3fb_Series - Metadata.csv', encoding = 'latin-1') #abrimos nuestro primer dataset

In [5]:
##Dimensiones del dataframe
print("numero de filas",  data1.shape[0])
print("numero de columnas",  data1.shape[1])

numero de filas 1091
numero de columnas 21


In [ ]:
data1.head(50) #visualizamos

Country Name Country Code         Series Name         Series Code  \
0           Afghanistan          AFG  Child malnutrition   5.51.01.02.malnut   
1           Afghanistan          AFG     Child mortality   5.51.01.03.mortal   
2           Afghanistan          AFG     Gender equality   5.51.01.07.gender   
3           Afghanistan          AFG        Immunization    5.51.01.04.immun   
4           Afghanistan          AFG   Population census   3.11.01.01.popcen   
5           Afghanistan          AFG      Poverty survey  5.14.01.01.povsurv   
6           Afghanistan          AFG    UNESCO reporting   5.12.01.01.unesco   
7               Albania          ALB  Child malnutrition   5.51.01.02.malnut   
8               Albania          ALB     Child mortality   5.51.01.03.mortal   
9               Albania          ALB     Gender equality   5.51.01.07.gender   
10              Albania          ALB        Immunization    5.51.01.04.immun   
11              Albania          ALB   Population census   3.11.01.01.popcen   
12              Albania          ALB      Poverty survey  5.14.01.01.povsurv   
13              Albania          ALB    UNESCO reporting   5.12.01.01.unesco   
14              Algeria          DZA  Child malnutrition   5.51.01.02.malnut   
15              Algeria          DZA     Child mortality   5.51.01.03.mortal   
16              Algeria          DZA     Gender equality   5.51.01.07.gender   
17              Algeria          DZA        Immunization    5.51.01.04.immun   
18              Algeria          DZA   Population census   3.11.01.01.popcen   
19              Algeria          DZA      Poverty survey  5.14.01.01.povsurv   
20              Algeria          DZA    UNESCO reporting   5.12.01.01.unesco   
21               Angola          AGO  Child malnutrition   5.51.01.02.malnut   
22               Angola          AGO     Child mortality   5.51.01.03.mortal   
23               Angola          AGO     Gender equality   5.51.01.07.gender   
24               Angola          AGO        Immunization    5.51.01.04.immun   
25               Angola          AGO   Population census   3.11.01.01.popcen   
26               Angola          AGO      Poverty survey  5.14.01.01.povsurv   
27               Angola          AGO    UNESCO reporting   5.12.01.01.unesco   
28  Antigua and Barbuda          ATG  Child malnutrition   5.51.01.02.malnut   
29  Antigua and Barbuda          ATG     Child mortality   5.51.01.03.mortal   
30  Antigua and Barbuda          ATG     Gender equality   5.51.01.07.gender   
31  Antigua and Barbuda          ATG        Immunization    5.51.01.04.immun   
32  Antigua and Barbuda          ATG   Population census   3.11.01.01.popcen   
33  Antigua and Barbuda          ATG      Poverty survey  5.14.01.01.povsurv   
34  Antigua and Barbuda          ATG    UNESCO reporting   5.12.01.01.unesco   
35            Argentina          ARG  Child malnutrition   5.51.01.02.malnut   
36            Argentina          ARG     Child mortality   5.51.01.03.mortal   
37            Argentina          ARG     Gender equality   5.51.01.07.gender   
38            Argentina          ARG        Immunization    5.51.01.04.immun   
39            Argentina          ARG   Population census   3.11.01.01.popcen   
40            Argentina          ARG      Poverty survey  5.14.01.01.povsurv   
41            Argentina          ARG    UNESCO reporting   5.12.01.01.unesco   
42              Armenia          ARM  Child malnutrition   5.51.01.02.malnut   
43              Armenia          ARM     Child mortality   5.51.01.03.mortal   
44              Armenia          ARM     Gender equality   5.51.01.07.gender   
45              Armenia          ARM        Immunization    5.51.01.04.immun   
46              Armenia          ARM   Population census   3.11.01.01.popcen   
47              Armenia          ARM      Poverty survey  5.14.01.01.povsurv   
48              Armenia          ARM    UNESCO reporting   5.12.01.01.unesco   
49         

podemos observas que en las columnas se podria hacer lo siguiente:

1) Dropear la columna 'Series Code' ya que no aporta nada al modelo que deseamos implementar

2) Las columnas que que contienen los diferentes años normalizarlos de: ("ANIO"[YR"ANIO"] a solo "anio")

3) Cambiar el nombre de las primeras 3 columnas a solo minusculas

In [16]:
 #revizamos si hay valores nulos
data1.isnull().sum()

Country Name      3
Country Code      5
Series Name       5
Series Code       5
2004 [YR2004]     5
2005 [YR2005]     5
2006 [YR2006]     5
2007 [YR2007]    13
2008 [YR2008]    13
2009 [YR2009]    13
2010 [YR2010]    13
2011 [YR2011]    13
2012 [YR2012]    13
2013 [YR2013]    13
2014 [YR2014]    13
2015 [YR2015]    13
2016 [YR2016]    13
2017 [YR2017]    13
2018 [YR2018]    13
2019 [YR2019]    13
2020 [YR2020]    13
dtype: int64

In [17]:
((data1.isnull().sum())*100 / len(data1))

Country Name     0.274977
Country Code     0.458295
Series Name      0.458295
Series Code      0.458295
2004 [YR2004]    0.458295
2005 [YR2005]    0.458295
2006 [YR2006]    0.458295
2007 [YR2007]    1.191567
2008 [YR2008]    1.191567
2009 [YR2009]    1.191567
2010 [YR2010]    1.191567
2011 [YR2011]    1.191567
2012 [YR2012]    1.191567
2013 [YR2013]    1.191567
2014 [YR2014]    1.191567
2015 [YR2015]    1.191567
2016 [YR2016]    1.191567
2017 [YR2017]    1.191567
2018 [YR2018]    1.191567
2019 [YR2019]    1.191567
2020 [YR2020]    1.191567
dtype: float64

In [23]:
(data1.isnull().sum()).sum()

215

In [34]:
nulos = (data1.isnull().sum()).sum() / 38185 * 100
print(f'Porcentaje total de valores nulos presentes en el dataframe {nulos}% (representados en una escala de 0-100%)')

Porcentaje total de valores nulos presentes en el dataframe 0.5630483174021212% (representados en una escala de 0-100%)


In [ ]:
data1["Series Name"].unique() ##chequeamos para ver si hay inconsistencias en la columna Series Name

array(['Child malnutrition', 'Child mortality', 'Gender equality',
       'Immunization', 'Population census', 'Poverty survey',
       'UNESCO reporting'], dtype=object)

In [14]:
data1.tail()

Country Name                                       Country Code  \
1086   5.51.01.07.gender  Gender equality in education (gross enrollment...   
1087    5.51.01.04.immun  Child immunization (proportion of one-year-old...   
1088   3.11.01.01.popcen                                  Population Census   
1089  5.14.01.01.povsurv                                    Poverty surveys   
1090   5.12.01.01.unesco                      Enrolment reporting to UNESCO   

                                            Series Name  \
1086  The indicator is defined as the ratio of the g...   
1087  The proportion of one-year-old children immuni...   
1088  Population censuses collect data on the size, ...   
1089  Poverty surveys collect data on household inco...   
1090  UNESCO Institute of Statistics compiles data o...   

                                            Series Code  \
1086  The indicator is defined as the ratio of the g...   
1087  The proportion of one-year-old children immuni...   
1088  Population censuses collect data on the size, ...   
1089  Poverty surveys collect data on household inco...   
1090  UNESCO Institute of Statistics compiles data o...   

                                          2004 [YR2004]  \
1086  World Development Indicator (WDI) databank. Or...   
1087  World Development Indicator (WDI) databank. Or...   
1088  World Bank Microdata library. Original source:...   
1089                       World Bank Microdata library   
1090  World Bank EdStats. Original source: UNESCO In...   

                             2005 [YR2005]  \
1086  Statistical Capacity Indicator (SCI)   
1087  Statistical Capacity Indicator (SCI)   
1088  Statistical Capacity Indicator (SCI)   
1089  Statistical Capacity Indicator (SCI)   
1090  Statistical Capacity Indicator (SCI)   

                                          2006 [YR2006] 2007 [YR2007]  \
1086  Scores are 1, 2/3, and 1/3 if the indicator is...           NaN   
1087  Score is 1 if the periodicity of the indicator...           NaN   
1088  Score is 1 if the country had a census at leas...           NaN   
1089  Scores are 1, and 1/2 if a survey is conducted...           NaN   
1090  Score is 1 if the country reported at least 3 ...           NaN   

     2008 [YR2008] 2009 [YR2009]  ... 2011 [YR2011] 2012 [YR2012]  \
1086           NaN           NaN  ...           NaN           NaN   
1087           NaN           NaN  ...           NaN           NaN   
1088           NaN           NaN  ...           NaN           NaN   
1089           NaN           NaN  ...           NaN           NaN   
1090           NaN           NaN  ...           NaN           NaN   

     2013 [YR2013] 2014 [YR2014] 2015 [YR2015] 2016 [YR2016] 2017 [YR2017]  \
1086           NaN           NaN           NaN           NaN           NaN   
1087           NaN           NaN           NaN           NaN           NaN   
1088           NaN           NaN           NaN           NaN           NaN   
1089           NaN           NaN           NaN           NaN           NaN   
1090           NaN           NaN           NaN           NaN           NaN   

     2018 [YR2018] 2019 [YR2019] 2020 [YR2020]  
1086           NaN           NaN           NaN  
1087           NaN           NaN           NaN  
1088           NaN           NaN           NaN  
1089           NaN           NaN           NaN  
1090           NaN           NaN           NaN  

[5 rows x 21 columns]

In [ ]:
data1['Country Name'].unique()

array(['Afghanistan', 'Albania', 'Algeria', 'Angola',
       'Antigua and Barbuda', 'Argentina', 'Armenia', 'Azerbaijan',
       'Bangladesh', 'Belarus', 'Belize', 'Benin', 'Bhutan', 'Bolivia',
       'Bosnia and Herzegovina', 'Botswana', 'Brazil', 'Bulgaria',
       'Burkina Faso', 'Burundi', 'Cabo Verde', 'Cambodia', 'Cameroon',
       'Central African Republic', 'Chad', 'Chile', 'China', 'Colombia',
       'Comoros', 'Congo, Dem. Rep.', 'Congo, Rep.', 'Costa Rica',
       "Cote d'Ivoire", 'Croatia', 'Djibouti', 'Dominica',
       'Dominican Republic',
       'East Asia & Pacific (excluding high income)', 'Ecuador',
       'Egypt, Arab Rep.', 'El Salvador', 'Equatorial Guinea', 'Eritrea',
       'Eswatini', 'Ethiopia',
       'Europe & Central Asia (excluding high income)', 'Fiji', 'Gabon',
       'Gambia, The', 'Georgia', 'Ghana', 'Grenada', 'Guatemala',
       'Guinea', 'Guinea-Bissau', 'Guyana', 'Haiti', 'Honduras',
       'IBRD only', 'IDA & IBRD total', 'IDA total', 'India', 'In

In [22]:
# Contamos con un total de 164 paises
data1['Country Name'].nunique()

164

In [ ]:
for item in data1["Country Name"].unique():
    print(item, end="___")

Afghanistan___Albania___Algeria___Angola___Antigua and Barbuda___Argentina___Armenia___Azerbaijan___Bangladesh___Belarus___Belize___Benin___Bhutan___Bolivia___Bosnia and Herzegovina___Botswana___Brazil___Bulgaria___Burkina Faso___Burundi___Cabo Verde___Cambodia___Cameroon___Central African Republic___Chad___Chile___China___Colombia___Comoros___Congo, Dem. Rep.___Congo, Rep.___Costa Rica___Cote d'Ivoire___Croatia___Djibouti___Dominica___Dominican Republic___East Asia & Pacific (excluding high income)___Ecuador___Egypt, Arab Rep.___El Salvador___Equatorial Guinea___Eritrea___Eswatini___Ethiopia___Europe & Central Asia (excluding high income)___Fiji___Gabon___Gambia, The___Georgia___Ghana___Grenada___Guatemala___Guinea___Guinea-Bissau___Guyana___Haiti___Honduras___IBRD only___IDA & IBRD total___IDA total___India___Indonesia___Iran, Islamic Rep.___Iraq___Jamaica___Jordan___Kazakhstan___Kenya___Kiribati___Kosovo___Kyrgyz Republic___Lao PDR___Latin America & Caribbean (excluding high income)

Evaluando una a una las diferentes regiones presentes en el dataset se deberia hacer una normalizacion en las siguientes regiones:

Fed. Sts a solo Micronesia ya que pertenece a regiones de Micronesia.

Congo, Dem. Rep. y Congo, Rep. a solo República Democrática del Congo y Republic of Congo respectivamente para evitar confusiones ya que son naciones que comparten el mismo nombre distintos

la columna 'Series Code' en caso de querer usarse mas adelante se deberian cambiar todas las letras a minusculas

En las columnas que contienen valores por años se pudo apreciar que:

Existen valores de la forma ".." dando como constancia de que este año nu hubo registro de dicho nombre de serie para determinado país, se recomienda reemplazar dicho valor por un 0

En aquellos años donde los datos son del tipo decimal se recomienda redondear a 2 decimales.

In [8]:
j = []
for i in data1.iloc[:,4:35].columns:
    j.append(len(data1[data1[i] == '..']))
print('Numero total de valores faltantes: ' + str(np.sum(j) ))

Numero total de valores faltantes: 1589


In [10]:
data1.shape

(1091, 21)

In [11]:
print('Numero total de valores: ' + str(1091* 35))

Numero total de valores: 38185


In [12]:
print('Porcentaje de valores faltantes: ' + str(( 1589 / 38185) * 100) + '%')

Porcentaje de valores faltantes: 4.161319890009166%


+ Presentamos un total de 215 datos nulos, en terminos de porcentajes representa un 0.5% (en una escala de porcentaje de 0-100)

+ Tambien podemos apreciar que presentamos un porcentaje muy bajo de datos faltantes, el cual es un 4%


ACLARACION: 
            
            Valores nulos = NaN

            Valores faltantes = '..' 